docs/tutorial/go.rst

```diff
@@ -0,0 +1,351 @@
+
+Pack the rock:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:pack]
+    :end-before: [docs:pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on your network, this step can take a couple of minutes to finish.
+
+    ``ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`` is required whilst the Go
+    extension is experimental.
+
+Once Rockcraft has finished packing the Go rock, you'll find a new file in
+your working directory (an `OCI <OCI_image_spec_>`_ archive) with the ``.rock``
+extension:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:ls-rock]
+    :end-before: [docs:ls-rock-end]
+    :dedent: 2
+
+.. note::
+    If you changed the ``name`` or ``version`` in ``rockcraft.yaml`` or are not
+    on an ``amd64`` platform, the name of the ``.rock`` file will be different
+    for you.
+
+    By default, the ``go-framework`` will use the ``bare`` base.
+    The ``bare`` base is recommended for production applications because
+    the image size will be smaller, and the risk of security issues lower.
+    However, you may want to change the base to ``ubuntu@24.04`` during the
+    development process to benefit from the extra utilities provided.
+
+
+Run the Go rock with Docker
+===========================
+
+
+We already have the rock as an `OCI <OCI_image_spec_>`_ archive. Now we
+need to load it into Docker:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+
+Check that the image was successfully loaded into Docker:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:docker-images]
+    :end-before: [docs:docker-images-end]
+    :dedent: 2
+
+The output should list the Go container image, along with its tag, ID and
+size:
+
+..  code-block:: text
+    :class: log-snippets
+
+    REPOSITORY       TAG       IMAGE ID       CREATED         SIZE
+    go-hello-world   0.1       f3abf7ebc169   5 minutes ago   15.7MB
+
+.. note::
+    The size of the image reported by Docker is the uncompressed size which is
+    larger than the size of the compressed ``.rock`` file.
+
+Now we're finally ready to run the rock and test your containerised Go
+application:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:docker-run]
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the Go
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/go/task.yaml
+    :language: text
+    :start-after: [docs:curl-go-rock]
+    :end-before: [docs:curl-go-rock-end]
+    :dedent: 2
+
+The Go application should again respond with ``Hello, world!``.
+
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying the Go rock, you can always get the application logs via
+``pebble``:
```

Comment:
This is the very first occurrence of "pebble" in the whole rockcraft doc website: It doesn't show up on the home page, it doesn't show up in the first two tutorials, either.

Following the doc's structure, I would read the homepage, follow the first two tutorials, then follow the Go tutorial because other languages don't interest me, and bam, I met the term "pebble", and it's confusing: What is pebble?

Although there is a link, but again, this adds cognitive load for a tutorial that is supposed to be simple, as simple as building and running an OCI image.

Even if the user is already familiar with pebble or switched to the pebble doc and then back, they would still have a question mark: I already built an executable binary, why can't I just add it in the OCI image and run it, why would I need a service manager for a single executable?

Reply:
I think we can link to information about pebble, I don't think we should get into the details of explaining pebble here. We should update the rest of the tutorials as well

Reply:
I updated the tutorial so the first reference to pebble has the link. I do not think pebble will be a hurdle for the user here, as the user can just gloss over it (and run the command and see the logs). It they want to have more information about it, there is no other options except reading the docs (and explaining it here will make the tutorial complex).